### PR TITLE
[3079] improve(client-java): add equals method to GravitinoVersion

### PR DESCRIPTION
**Pull request by argel6767**
### What changes were proposed in this pull request?

creating an .equals() method in GravitinoVersion.java as proposed in #3079 

### Why are the changes needed?
to compensate for potential different results in different versions of Java depending on whether compareTo() or equals() is used

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Added tests in TestGravitinoVersion.java

### CODE REVIEW

**Comment by justinmclean** [comment]
It looks like you may have added some extra unrelated files in this PR? The build is failing because those files don't have the correct file license headers.

File: catalogs/catalog-hive/bin/generated-sources/annotations/com/datastrato/gravitino/catalog/hive/ImmutableConfElement.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+ * Use the builder to create immutable instances:
+ * {@code ImmutableConfElement.builder()}.
+ */
+@Generated(from = "CachedClientPool.ConfElement", generator = "Immutables")
+@SuppressWarnings({"all"})
+@ParametersAreNonnullByDefault
+@javax.annotation.Generated("org.immutables.processor.ProxyProcessor")
+@Immutable
+@CheckReturnValue
+final class ImmutableConfElement
```
Comment (jerryshao) [inline catalogs/catalog-hive/bin/generated-sources/annotations/com/datastrato/gravitino/catalog/hive/ImmutableConfElement.java:28]: Seems these codes are unrelated, can you please remove these codes?

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoVersion.java [hunk 2]
```diff
@@ -56,4 +56,23 @@ public int compareTo(Object o) {
...
+  @Override
+    public boolean equals(Object object) {
+      if (!(object instanceof GravitinoVersion)) {
+        return false;
+      }
+  
+      GravitinoVersion othVersion = (GravitinoVersion) object;
+  
+      int[] versionNum = getVersionNumber();
+      int[] othVersionNum = othVersion.getVersionNumber();
```
Comment (jerryshao) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoVersion.java:69]: I would suggest change `oth` to `other` here and above.

File: clients/client-java/src/test/java/com/datastrato/gravitino/client/TestGravitinoVersion.java [hunk 3]
```diff
@@ -70,4 +71,22 @@ void testVersionCompare() {
     version2 = new GravitinoVersion("2.5.3-SNAPSHOT", "2023-01-01", "1234567");
     assertEquals(0, version1.compareTo(version2));
   }
+  @Test
```
Comment (jerryshao) [inline clients/client-java/src/test/java/com/datastrato/gravitino/client/TestGravitinoVersion.java:74]: Can you please add a blank line above?

File: clients/client-java/src/test/java/com/datastrato/gravitino/client/TestGravitinoVersion.java [hunk 4]
```diff
@@ -70,4 +71,22 @@ void testVersionCompare() {
...
+      version1 = new GravitinoVersion("3.1.4", "2022-01-01", "1234567");
+      version2 = new GravitinoVersion("3.1.2", "2022-01-01", "1234567");
+      assertFalse(version1.equals(version2));
+  
+      //test equal with suffix
+      version1 = new GravitinoVersion("3.1.6-BETA", "2022-01-01", "1234567");
+      version2 = new GravitinoVersion("3.1.6", "2022-01-01", "1234567");
+      assertTrue(version1.equals(version2));
+    }
+
```
Comment (jerryshao) [inline clients/client-java/src/test/java/com/datastrato/gravitino/client/TestGravitinoVersion.java:91]: Can you please remove this empty line?

File: clients/client-java/src/test/java/com/datastrato/gravitino/client/TestGravitinoVersion.java [hunk 5]
```diff
@@ -70,4 +71,22 @@ void testVersionCompare() {
...
+      assertTrue(version1.equals(version2));
+  
+      //test not equal
+      version1 = new GravitinoVersion("3.1.4", "2022-01-01", "1234567");
+      version2 = new GravitinoVersion("3.1.2", "2022-01-01", "1234567");
+      assertFalse(version1.equals(version2));
+  
+      //test equal with suffix
+      version1 = new GravitinoVersion("3.1.6-BETA", "2022-01-01", "1234567");
+      version2 = new GravitinoVersion("3.1.6", "2022-01-01", "1234567");
```
Comment (jerryshao) [inline clients/client-java/src/test/java/com/datastrato/gravitino/client/TestGravitinoVersion.java:88]: I don't think the versions "3.1.6" and "3.1.6-BETA" are the same. The way you do comparison using parsed version number integer array seems not correct.
